frontend/src/pages/projects/screens/detail/ProjectActions.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { Dropdown, DropdownItem, MenuToggle, DropdownList } from '@patternfly/react-core';
+import { useNavigate } from 'react-router-dom';
+import { getDashboardMainContainer } from '~/utilities/utils';
+import { AccessReviewResourceAttributes, ProjectKind } from '~/k8sTypes';
+import { useAccessReview } from '~/api';
+import DeleteProjectModal from '~/pages/projects/screens/projects/DeleteProjectModal';
+import ManageProjectModal from '~/pages/projects/screens/projects/ManageProjectModal';
+
+const projectEditAccessReview: AccessReviewResourceAttributes = {
+  group: 'project.openshift.io',
+  resource: 'projectrequests',
+  verb: 'update',
+};
+
+const projectDeleteAccessReview: AccessReviewResourceAttributes = {
+  group: 'project.openshift.io',
+  resource: 'projectrequests',
+  verb: 'delete',
+};
+
+type Props = {
+  project: ProjectKind;
+};
+
+const ProjectActions: React.FC<Props> = ({ project }) => {
+  const navigate = useNavigate();
+  const [canEdit, editRbacLoaded] = useAccessReview({
+    ...projectEditAccessReview,
+    namespace: project.metadata.name,
+  });
+  const [canDelete, deleteRbacLoaded] = useAccessReview({
+    ...projectDeleteAccessReview,
+    namespace: project.metadata.name,
+  });
+  const [open, setOpen] = React.useState(false);
+  const [deleteOpen, setDeleteOpen] = React.useState(false);
+  const [editOpen, setEditOpen] = React.useState(false);
+
+  if (!editRbacLoaded || !deleteRbacLoaded || (!canEdit && !canDelete)) {
+    return null;
+  }
+
+  const DropdownComponent = (
+    <Dropdown
+      onOpenChange={setOpen}
+      onSelect={() => setOpen(false)}
+      toggle={(toggleRef) => (
+        <MenuToggle
+          aria-label="Actions"
+          data-testid="project-actions"
+          variant="secondary"
+          ref={toggleRef}
+          onClick={() => {
+            setOpen(!open);
+          }}
+        >
+          Actions
+        </MenuToggle>
+      )}
+      isOpen={open}
+      popperProps={{ position: 'right', appendTo: getDashboardMainContainer() }}
+    >
+      <DropdownList>
```

Comment:
Based on the mocks, should there be a divider between these "Edit" and "Delete" dropdown items?

Reply:
@jenny-s51 - We don't have design guidelines for action menus yet. When I look at other areas of the product, there are pages in the dashboard that consistently display a divider above Delete. But there are places where we have just Edit and Delete in a menu without a divider (Workbenches and Permissions list items in the Project details page). 

The only guidelines I see on this topic within PF are under [Menu > Red text menu](https://www.patternfly.org/components/menus/menu/design-guidelines#red-text-menu):
> A divider should be used to separate the destructive menu items from the non-destructive items.

We aren't using this variant on the Delete menu action (and maybe we should consider that when we draft guidelines), but the point about separating destructive menu items from the non-destructive menu items is the piece that I think is relevant here. I think this recommendation should apply regardless of how many menu items there are. So I am in favor of implementing the UI as designed in this case.

Reply:
@jenny-s51 @jgiardino Looking around the UI, I see:

- Connections (recently added) does not have the separator.
- Data connections does not have the separator
- Cluster storage does not have the separator
- Permissions (both user and group) do not have the separator
- Workbenches does not have the separator
- Pipeline servers does have the separator
- Pipeline versions does have the separator

I can add it here but it would be mostly inconsistent. I would suggest UX file an issue to make all of the menus consistent and we address all of them at once.

Reply:
@jeff-phillips-18 @jenny-s51 @jgiardino I agree with Jeff. I’d prefer to create a separate issue once we have the menu guidelines ready to update the styling for all menus at once and ensure consistency. For now, I’d vote to drop the divider from the implementation of the Project Details action.

Reply:
Thank you for your clarity here @jgiardino. Filing a separate ticket to make all of these menus consistent sounds like a good plan @jeff-phillips-18 @simrandhaliw.